src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -13,6 +13,9 @@ public class StringFunctions {
     private static final char SPACE = ' ';
 
     public static final String reverseWithStringConcat(String string) {
+        if (string == null || string.isEmpty()) {
```

Comment:
Added a check to return an empty string if the input is null or empty to prevent exceptions and handle edge cases gracefully.